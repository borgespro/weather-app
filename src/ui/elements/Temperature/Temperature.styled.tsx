import styled from 'styled-components/native';

import { CustomThemeProps } from '@/typings/theme';

export const CustomText = styled.Text<CustomThemeProps>`
  color: ${(props) => props.theme.colors.text};
  font-size: ${(props) => props.theme.typography.sizes.xxxl};
  font-family: ${(props) => props.theme.typography.fonts.montserrat?.medium};
`;
