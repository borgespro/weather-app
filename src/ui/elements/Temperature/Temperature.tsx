import React from 'react';

import { CustomText } from './Temperature.styled';

export type TemperatureProps = {
  degree: number;
};

export default function Temperature({ degree = 0 }: TemperatureProps): React.ReactElement {
  return <CustomText>{` ${degree.toFixed()}ยบ`}</CustomText>;
}
