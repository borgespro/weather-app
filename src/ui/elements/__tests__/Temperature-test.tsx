import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Temperature from '../Temperature';

describe('Temperature Element', () => {
  it('should render without issues', () => {
    const component = shallow(<Temperature degree={30.5} />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
    const label = component.text();
    expect(label).toEqual(' 31ยบ');
  });
});
