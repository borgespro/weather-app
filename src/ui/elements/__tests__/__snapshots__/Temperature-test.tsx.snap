// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature Element should render without issues 1`] = `
<Styled(Text)>
   31ยบ
</Styled(Text)>
`;
